app/stac_api/pgtriggers.py

```diff
@@ -61,79 +80,100 @@ def generate_child_triggers(parent_name, child_name):
     ]
 
 
-def generates_asset_triggers():
-    '''Generates Asset triggers
+def asset_counter_trigger(count_table, value_field):
+    '''Creates triggers to adjust the asset summary on a counter table.
 
-    Those triggers act on `insert`, `update` and `delete` Asset event and do the followings:
-      - Update the `updated` and `etag` fields of the assets and their parents
-      - Update the parent collection summaries.
-      - Update the parent item `update_interval` by using the minimal aggregation of all of its
-        assets
+    Args:
+         count_table: tha table name to be updated (without prefix stac_api_)
+         value_field: summary field name on the asset
 
-    Returns: tuple
-        tuple for all needed triggers
+    Returns:
+        List of triggers
     '''
 
-    class UpdateCollectionSummariesTrigger(pgtrigger.Trigger):
+    class UpdateDeleteCounterTrigger(pgtrigger.Trigger):
         when = pgtrigger.After
-        declare = [
-            ('asset_instance', 'stac_api_asset%ROWTYPE'),
-            ('related_collection_id', 'INT'),
-            ('collection_summaries', 'RECORD'),
-        ]
-        func = '''
-        asset_instance = COALESCE(NEW, OLD);
+        declare = [('asset_instance', 'stac_api_asset%ROWTYPE'), ('related_collection_id', 'INT')]
+        func = f'''
+        asset_instance = OLD;
 
         related_collection_id = (
             SELECT collection_id FROM stac_api_item
             WHERE id = asset_instance.item_id
         );
 
-        -- Update related item auto variables
-        UPDATE stac_api_item SET
-            updated = now(),
-            etag = gen_random_uuid()
-        WHERE id = asset_instance.item_id;
+        UPDATE stac_api_{count_table}
+        SET count = count-1
```

Comment:
if count-1 will be 0 the entry could be removed from the `stac_api_{count_table}`

Reply:
Yes, I forgot to look at that. I changed it so now the entry should be removed if count is 0. Thanks!